plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetastoreClosure.java

```diff
@@ -98,10 +98,10 @@ public void updateTableStatistics(HiveIdentity identity, String databaseName, St
         delegate.updateTableStatistics(identity, databaseName, tableName, update, transaction);
     }
 
-    public void updatePartitionStatistics(HiveIdentity identity, String databaseName, String tableName, String partitionName, Function<PartitionStatistics, PartitionStatistics> update)
+    public void updatePartitionStatistics(HiveIdentity identity, String databaseName, String tableName, String partitionName, AcidTransaction transaction, Function<PartitionStatistics, PartitionStatistics> update)
```

Comment:
Any reason not to use the same parameter order as for `updateTableStatistics` above: `update, transaction` ?

Reply:
I'd prefer an ordering of arguments that place the Function, which is usually a lambda, as the last arg, because other arguments aren't obvious if they are after a lambda.  They should be consistent, so I changed `updateTableStatistics` to be consistent with the changed `updatePartitionStatistics`.  I meant to do that in this PR but I spaced.

Reply:
> so I changed updateTableStatistics to be consistent

ideally, this should be a separate, preparatory commit.

(it may also go into separate PR for faster merge)

Reply:
I extracted it as the first commit of the PR.